### PR TITLE
Register as participant

### PR DESCRIPTION
**Pull request by dmnk89**

### CODE REVIEW

File: src/app/admin/participants/participant.model.ts [hunk 1]
```diff
@@ -11,4 +12,16 @@ export class Participant {
   creationDate: string;
   registrationDate: string;
   ticketSendDate: string;
+
+  public constructor(obj: any = {}) {
+    Object.assign(this, obj);
+  }
+}
+
+export class Voucher {
```
Comment (margielm) [inline src/app/admin/participants/participant.model.ts]: lets move it to a separate file

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 2]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+      <mat-form-field>
+        <mat-select name="size" id="size" formControlName="size" required placeholder="t-shirt size">
+          <mat-option></mat-option>
+          <mat-option value="S">S</mat-option>
+          <mat-option value="M">M</mat-option>
+          <mat-option value="L">L</mat-option>
+          <mat-option value="XL">XL</mat-option>
+          <mat-option value="XXL">XXL</mat-option>
+        </mat-select>
+        <mat-error *ngIf="size.invalid && (size.dirty || size.touched || submitted)">
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: those additional checks are not needed in angular Material Design. It has very sensible dedaults.

so we just need `*ngIf='side.invalid'`

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 3]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+        <mat-select name="experience" id="experience" formControlName="experience" required placeholder="your experience">
+          <mat-option></mat-option>
+          <mat-option value="I am just learning">I am just learning</mat-option>
+          <mat-option value="Less than a year">Less than a year</mat-option>
+          <mat-option value="1-2 years">1-2 years</mat-option>
+          <mat-option value="2-4 years">2-4 years</mat-option>
+          <mat-option value="4-8 years">4-8 years</mat-option>
+          <mat-option value="Over 8 years">Over 8 years</mat-option>
+        </mat-select>
+        <mat-error *ngIf="experience.invalid && (experience.dirty || experience.touched || submitted)">
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: same here.

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 4]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+          <mat-option></mat-option>
+          <mat-option value="student">Student</mat-option>
+          <mat-option value="Developer">Developer</mat-option>
+          <mat-option value="Team Leader">Team Leader</mat-option>
+          <mat-option value="Project Leader">Project Leader</mat-option>
+          <mat-option value="Architect">Architect</mat-option>
+          <mat-option value="Tester">Tester</mat-option>
+          <mat-option value="Other">Other</mat-option>
+        </mat-select>
+        <mat-error *ngIf="role.invalid && (role.dirty || role.touched || submitted)">
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: and here :)

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 5]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+      <mat-form-field>
+        <mat-select name="role" id="role" formControlName="role" required placeholder="your role" >
+          <mat-option></mat-option>
+          <mat-option value="student">Student</mat-option>
+          <mat-option value="Developer">Developer</mat-option>
+          <mat-option value="Team Leader">Team Leader</mat-option>
+          <mat-option value="Project Leader">Project Leader</mat-option>
+          <mat-option value="Architect">Architect</mat-option>
+          <mat-option value="Tester">Tester</mat-option>
+          <mat-option value="Other">Other</mat-option>
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: I am wondering if we can ask this data now (GDPR). We dont need this data to do our business, so for me it is questionable.

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 6]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+      </mat-form-field>
+      <mat-form-field>
+        <mat-select name="experience" id="experience" formControlName="experience" required placeholder="your experience">
+          <mat-option></mat-option>
+          <mat-option value="I am just learning">I am just learning</mat-option>
+          <mat-option value="Less than a year">Less than a year</mat-option>
+          <mat-option value="1-2 years">1-2 years</mat-option>
+          <mat-option value="2-4 years">2-4 years</mat-option>
+          <mat-option value="4-8 years">4-8 years</mat-option>
+          <mat-option value="Over 8 years">Over 8 years</mat-option>
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: I am wondering if we can ask this data now (GDPR). We dont need this data to do our business, so for me it is questionable.

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 7]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+      <mat-form-field class="full-width">
+        <input matInput placeholder="voucher" type="text" id="voucher2"
+             name="voucher" formControlName="voucher"
+             maxlength="255">
+      </mat-form-field>
+      <mat-form-field>
+        <input type="text" id="city" placeholder="city" formControlName="city"
+             matInput name="city"
+             required maxlength="255">
+        {{registrationForm['city']|json}}
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: looks like debug string :)

File: src/app/pages/registration/registration-form/registration-form.component.html [hunk 8]
```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
...
+    <form (ngSubmit)="save()" [formGroup]="registrationForm" *ngIf="registrationForm">
+      <mat-form-field class="full-width">
+        <input matInput placeholder="voucher" type="text" id="voucher2"
+             name="voucher" formControlName="voucher"
+             maxlength="255">
+      </mat-form-field>
+      <mat-form-field>
+        <input type="text" id="city" placeholder="city" formControlName="city"
+             matInput name="city"
+             required maxlength="255">
```
Comment (margielm) [inline src/app/pages/registration/registration-form/registration-form.component.html]: I am wondering if we can ask this data now (GDPR). We dont need this data to do our business, so for me it is questionable.

Comment (dmnk89) [reply src/app/pages/registration/registration-form/registration-form.component.html]: lets talk about it on todays call

**Pull request merged**